RealmSwift/AsymmetricObject.swift

```diff
@@ -0,0 +1,105 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Foundation
+import Realm
+import Realm.Private
+
+/**
+ `AsymmetricObject` is a base class used to define asymmetric Realm objects.
+
+ Asymmetric objects can only be created using the `create(_ object:)`
+ function, and cannot be added, removed or queried.
+ When created asymmetric objects will synced unidirectionally to the  MongoDB
+ database and cannot be accessed locally.
+
+ Incoming links from any asymmetric table are not allowed, meaning embedding
+ an asymmetric object within an `Object` will throw an error.
+
+ The property types supported on `AsymmetricObject` are the same as for `Object`,
+ except for that asymmetric objects can only link to embedded objects, so `Object`
+ and `List<Object>` properties are not supported (`EmbeddedObject` and
+ `List<EmbeddedObject>` *are*).
+
+ ```swift
+ class Person: AsymmetricObject {
+     @Persisted(primaryKey: true) var _id: ObjectId = ObjectId.generate()
+     @Persisted var name: String
+     @Persisted var age: Int
+ }
+ ```
+ */
+public typealias AsymmetricObject = RealmSwiftAsymmetricObject
+extension AsymmetricObject: _RealmCollectionValueInsideOptional {
+    /// :nodoc:
+    public class override final func isAsymmetric() -> Bool {
```

Comment:
I don't see the need to override this getter?

Reply:
we need to do this for schema discovery, only asymmetric objects (object that inherit from `AsymmetricObject`), will have this set as true, and when we are getting the object schema for the specific class https://github.com/realm/realm-swift/blob/e6b763cf9bcf2359c9c423336e08aee277f04635/Realm/RLMObjectSchema.mm#L144, only the objects which have this true will be set as asymmetric, we set this flag for schema validation with the server and core's using this to allow/disallow some operation with asymmetric objects.